### DISCUSSION TITLE
Select edition from install.esd

### DISCUSSION THREAD

**Discussion by CyberC4p0** [ipxe/wimboot#37] [Q&A] [answered]
On the wimboot section of the [iPXE](https://ipxe.org/wimboot) website

![image](https://user-images.githubusercontent.com/67917738/202038610-eb11267e-9cad-41a2-9581-12e05ad45b38.jpeg)

It mentions that multi-image wim files will work by specifying the index of the version we want to boot. I mounted the Windows 10 ISO but instead of finding the install.wim file, I found the install.esd. Will this work regardless? If not, how can I go on obtaining the install.wim file?

![image](https://user-images.githubusercontent.com/67917738/202042489-5a3c8bd5-6a4a-47e8-bf27-7d1cf9553a30.png)

Thank you for the help provided :D

**Comment by NiKiZe** [comment]
You need boot,wim which contains the winpe system.
Once you have that system running it will need install.esd/wim to install to disk. - You should not boot install.x

**Reply by CyberC4p0** [reply]
> You need boot,wim which contains the winpe system.
> 
> Once you have that system running it will need install.esd/wim to install to disk. - You should not boot install.x

Thank you for this. This solved my issue. I will specify Index 1 for windows 10 home while still booting the boot.wim file. Then when it gets the install.esd, it will get the image located on index 1. Now I understand! Thank you 😄

**Reply by NiKiZe** [reply]
This is not the right place to go over the basics, See https://ipxe.org/howto/winpe (ignore any mention of bcd and boot.sdi which no longer is needed)

Quick info:
use wimboot to start boot.wim
use startnet.cmd or other technique to mount a directory with the source files (sources directory in the iso)
Run setup.exe from that mounted path.

I recommend to start the smaller part of the boot.wim (first index) which should not cotain the setup.exe etc. (again that should be run from the mounted path)

There is https://github.com/NiKiZe/wimboot-install which is heavy WIP (you will need to make personal changes, and understand php to use it in a good way, At this stage i will not provide any support for it.

**Reply by NiKiZe** [reply]
> > You need boot,wim which contains the winpe system.
> > Once you have that system running it will need install.esd/wim to install to disk. - You should not boot install.x
> 
> Thank you for this. This solved my issue. I will specify Index 1 for windows 10 home while still booting the boot.wim file. Then when it gets the install.esd, it will get the image located on index 1. Now I understand! Thank you 😄

No that is totally wrong. the index of boot.wim has nothing to do with any indexes in install.wim/esd

**Reply by CyberC4p0** [reply]
> > > You need boot,wim which contains the winpe system.
> 
> > > Once you have that system running it will need install.esd/wim to install to disk. - You should not boot install.x
> 
> > 
> 
> > Thank you for this. This solved my issue. I will specify Index 1 for windows 10 home while still booting the boot.wim file. Then when it gets the install.esd, it will get the image located on index 1. Now I understand! Thank you 😄
> 
> 
> 
> No that is totally wrong. the index of boot.wim has nothing to do with any indexes in install.wim/esd

So then why specify an index? I want to be able to load different indexes of the install.esd when installing to a disk.

![image](https://user-images.githubusercontent.com/67917738/202042489-5a3c8bd5-6a4a-47e8-bf27-7d1cf9553a30.png)

Something that will let me choose like the menu above.

**Reply by NiKiZe** [reply]
You might want to read about autounattend.xml

**Reply by oschroeder8** [reply]
An ESD file is not technically a WIM file, though they serve a similar purpose.

On the ISO you will find a `boot.wim` file. That WIM file contains the Windows Setup program which, when started, would allow you to deploy the images contained in the ESD file. But when using WIMBOOT, that ESD file will not be accessible because you didn't load the whole ISO, just the WIM file. That's why @NiKiZe is referring to a mounted path where the ESD file would be accessible (the remaining content of the ISO).

If the ESD file was like a WIM file and you then picked an image index, you would boot directly into a Windows installation, not the process of deploying the image itself.

Hope that clarifies things.

**Reply by CyberC4p0** [reply]
> An ESD file is not technically a WIM file, though they serve a similar purpose.
> 
> 
> 
> On the ISO you will find a `boot.wim` file. That WIM file contains the Windows Setup program which, when started, would allow you to deploy the images contained in the ESD file. But when using WIMBOOT, that ESD file will not be accessible because you didn't load the whole ISO, just the WIM file. That's why @NiKiZe is referring to a mounted path where the ESD file would be accessible (the remaining content of the ISO).
> 
> 
> 
> If the ESD file was like a WIM file and you then picked an image index, you would boot directly into a Windows installation, not the process of deploying the image itself.
> 
> 
> 
> Hope that clarifies things.

Ohh ok! Thanks for the clarification 🤗

**Comment by CyberC4p0** [comment, marked as answer]
To answer my own question...

I had to add a ei.cfg to bring up the Select Edition menu during installation.

![image](https://user-images.githubusercontent.com/67917738/202062232-722f6916-0de3-4d32-a812-e49236d77a4d.jpeg)
Credit: https://www.intowindows.com/how-to-select-pro-edition-while-installing-windows-10/
